post: how to get promoted (#17)
natikgadzhi/respawn-io: content/posts/how-to-get-promoted.md

---
created: 2025-04-11
modified: 2025-04-11
title: How to get promoted faster
excerpt: Make your own growth plans, brag lists, prepare promotion packages, and for the love of gawd start writing.
tags:
  - growth
  - lifting-others
  - engineering-leadership
---

# How to get promoted faster

> Companies generally suck at promoting you at _what feels to you as_ the right time. Either you're in a startup and promotions feel concervative, or you're in a FAANG company and they're overly political. /shrug

Author: ```suggestion
Companies generally suck at promoting you at _what feels to you as_ the right time. Either you're in a startup and promotions feel conservative, or you're in a FAANG company and they're overly political. /shrug
```

At junior levels, good companies _expect_ that you will grow within a certain time period. That's because your true impact unleashes when you level up and stay at the company. Your promotions are _expected_ and you have to show up and provide enough evidence to back them up.

## Take ownership of your career

<WithAside>
Your manager should generally ensure you have a _growth plan_, they should advocate for _great things you made_ and _make your impact visible_, and then they would put together a _promotion package_, that is mostly a fancy word for a document describing why they think you're eligible for the next promotion.

<Aside>
[[understand-the-context|Making your work visible]] is generally helpful in putting together a statement like “Look at me, I'm outperforming my level and need a promotion yesterday”.
</Aside>
</WithAside>

**But you can and should take ownership of your trajectory and growth.** Here are a few useful tools and artefacts to help you think through leveling up.

## The growth plan

> A good growth plan is a document that describes the list of behaviors and skills that you need to master in order to grow to the next milestone or level. **It's not a checklist**. You don't get the promotion when you checked every item on the list of things to learn. But it should give you an idea of your skill and behavior gap.

Author: ```suggestion
<WithAside>
A good growth plan is a document that describes the list of behaviors and skills that you need to master in order to grow to the next milestone or level. **It's not a checklist**. You don't get the promotion when you checked every item on the list of things to learn. Instead, the list is a guideline and a roadmap that should give you focus and direction, and inform your vision for your future promotion package (more on that below).
<Aside>
Patrick Nilan says: "not a checklist" comment hits it both ways. Just because a person "checks all the boxes" doesn't mean they're automatically going to be promoted, but also, just because a person may not "check all the boxes" doesn't mean they're not ready for promotion — it's more important to focus on the projects/work that have a business impact than to shirk responsibilities just to say "hey I did this thing that no one asked for."
</Aside>
</WithAside>
```

Your brag list is the easiest correct answer to “tell me about yourself” on an intro conversation with a recruiter from a company that makes your heart skip a beat.

## The promotion package

At some point, your manager decides that it's time to promote you. To get a promotion confirmed, they would need to put together a “promotion package” and get that approved. Each company has a slightly different process (promotion cycles with calibrations, ad-hoc promotions, etc).

The manager will have to convince their boss and their peers that you already perform at the next level, you're as good as other folks on that level, and your performance is sustainable. Promotion document usually provides evidence of your next-level performance, calibrates it against your peers at that level, and has some additional evidence and references from other folks at your company.

<WithAside>

> **Here's the twist: you can _envision_ your own promotion package as it will look 6-12 months from where you are.** You might not know a specific format that your manager would use, but you can write down a vision of what will be true in a year so that your manager would be able to get your promotion approved.

Reviewer: A couple thoughts on this:
- The 6-12 months out implies this would be a living document that should be regularly updated to 1) confirm goals were met and 2) update to add additional goals for 6-12 months from the update. In other words, it feels like there should be two versions of the doc: a "stable version" of the achievements that have been realized and a "release candidate" version which includes goals/unrealized achievements. 
- Also, I personally have a tough time envisioning 6-12 months down the road for a couple reasons -- but primarily a brag list is concrete/granular list of achievements, and a growth plan can be a bit more abstract. It feels like a promotion package should be concrete achievements that align with the growth plan. Should the promotion package include "placeholders" for realized achievements that fall under abstract goals of the growth plan? Do you agree with this approach?

For example, a line item that can be included in a growth plan should include concrete examples in the promotion package:
> **Communicating dependencies and risks:**
> - Working with other teams directly or with your manager to make sure you have the resources and attention you need for the project to succeed.
> - if/when scope changes, or new risks become known, panic LOUDLY and come with trade off of options of how to de-risk.

Possible future promo package excerpt based on the above:
> Patrick has consistently demonstrated strong ownership by clearly communicating project dependencies and risks, ensuring teams stayed aligned and resourced such as [example of cross-team collaboration on project XYZ]. Patrick proactively surfaced uncovered risks along with thoughtful trade-offs, helping to de-risk projects effectively. [Example of raising a risk and proposing solutions].

<Aside>
TODO: Link a template / example.
</Aside>

> </WithAside>

Author: ```suggestion
</WithAside>

The vision for your promotion is a living document. If you have enough trust with your manager, work on it with them, if not yet — it's your responsibility. But, it's a document that you update every month and clarify both your assumptions of what is worth a promotion, and how much closer to that did you get.
```
